pairing_bot.go

```diff
@@ -186,31 +192,120 @@ func (pl *PairingLogic) endofbatch(w http.ResponseWriter, r *http.Request) {
+		wasAtRCLastWeek := recursersList[i].currentlyAtRC
+
+		//If they were at RC last week but not this week then we assume they have graduated or otherwise left RC
+		//In that case we remove them from pairing bot so that inactive people do not get matched
+		//If people who have left RC still want to use pairing bot, we give them the option to resubscribe
+		if wasAtRCLastWeek && !isAtRCThisWeek {
+			var message string
+
+			err = pl.rdb.Delete(ctx, recurserID)
+			if err != nil {
+				log.Println(err)
+				message = fmt.Sprintf("Uh oh, I was trying to offboard you since it's the end of batch, but something went wrong. Consider messaging %v to let them know this happened.", owner)
+			} else {
+				log.Println("This user has been unsubscribed from pairing bot: ", recurserEmail)
+				message = offboardedMessage
+			}
+
+			err := pl.un.sendUserMessage(ctx, botPassword, recurserEmail, message)
+			if err != nil {
+				log.Printf("Error when trying to send offboarding message to %s: %s\n", recurserEmail, err)
+			}
+		}
+
+		recurser.currentlyAtRC = isAtRCThisWeek
+
+		if err = pl.rdb.Set(ctx, recurserID, recurser); err != nil {
+			log.Printf("Error encountered while update currentlyAtRC status for user: %s", recurserEmail)
+		}
+
+	}
+}
+
+/*
+Sends out a "Welcome to Pairing Bot" message to 397 Bridge during the second week of RC to introduce people to RC.
+
+We don't send this welcome message during the first week since it's a bit overwhelming with all of the orientation meetings
+and people haven't had time to think too much about their projects.
+*/
+func (pl *PairingLogic) welcome(w http.ResponseWriter, r *http.Request) {
+	// Check that the request is originating from within app engine
+	// https://cloud.google.com/appengine/docs/flexible/go/scheduling-jobs-with-cron-yaml#validating_cron_requests
+	if r.Header.Get("X-Appengine-Cron") != "true" {
+		http.NotFound(w, r)
+		return
+	}
+
+	ctx := r.Context()
+
+	accessToken, err := pl.adb.GetKey(ctx, "rc-accesstoken", "key")
+	if err != nil {
+		log.Printf("Something weird happened trying to read the RC API access token from the database: %s", err)
+	}
+
+	if pl.rcapi.isSecondWeekOfBatch(accessToken) {
+		ctx := r.Context()
+		botPassword, err := pl.adb.GetKey(ctx, "apiauth", "key")
 
-		err = pl.rdb.Delete(ctx, recurserID)
 		if err != nil {
-			log.Println(err)
-			message = fmt.Sprintf("Uh oh, I was trying to offboard you since it's the end of batch, but something went wrong. Consider messaging %v to let them know this happened.", owner)
-		} else {
-			log.Println("A user was offboarded because it's the end of a batch.")
-			message = offboardedMessage
+			log.Println("Something weird happened trying to read the auth token from the database")
 		}
 
-		err := pl.un.sendUserMessage(ctx, botPassword, recurserEmail, message)
+		streamMessage := getWelcomeMessage()
+
+		err = pl.sm.postToTopic(ctx, botPassword, streamMessage, "397 Bridge", "🍐🤖")
```

Comment:
I'm amazed that I was able to use the pear and robot emojis in the code and actually have them work 😂